### PR TITLE
Add ConvNeXT V2

### PR DESCRIPTION
**Pull request by alaradirik**
# What does this PR do?
Adds [ConvNeXT V2](https://arxiv.org/pdf/2301.00808.pdf) to transformers, including a backbone. ConvNeXT V2 features minimal changes to the `ConvNextLayer` and achieves an average 1% accuracy gain over ConvNext V1. Original repo is over [here](https://github.com/facebookresearch/ConvNeXt-V2).

- [X ] Upload ImageNet 1K fine-tuned models
- [x] Upload ImageNet 22K fine-tuned models
- [x] Update model cards
- [ ] Fix TF model bugs

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [X ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ X] Did you write any new necessary tests?

### CODE REVIEW

File: src/transformers/models/convnextv2/configuration_convnextv2.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    documentation from [`PretrainedConfig`] for more information.
+
+    Args:
+        num_channels (`int`, *optional*, defaults to 3):
+            The number of input channels.
+        patch_size (`int`, optional, defaults to 4):
+            Patch size to use in the patch embedding layer.
+        num_stages (`int`, optional, defaults to 4):
+            The number of stages in the model.
+        hidden_sizes (`List[int]`, *optional*, defaults to [96, 192, 384, 768]):
```
Comment (sgugger) [inline src/transformers/models/convnextv2/configuration_convnextv2.py:51]: List defaults should be in code blocks.

File: src/transformers/models/convnextv2/configuration_convnextv2.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    def inputs(self) -> Mapping[str, Mapping[int, str]]:
+        return OrderedDict(
+            [
+                ("pixel_values", {0: "batch", 1: "num_channels", 2: "height", 3: "width"}),
+            ]
+        )
+
+    @property
+    def atol_for_validation(self) -> float:
+        return 1e-5
```
Comment (sgugger) [inline src/transformers/models/convnextv2/configuration_convnextv2.py:135]: To remove.

File: src/transformers/models/convnextv2/modeling_convnextv2.py [hunk 3]
```diff
@@ -0,0 +1,589 @@
...
+        self.pwconv1 = nn.Linear(dim, 4 * dim)  # pointwise/1x1 convs, implemented with linear layers
+        self.act = ACT2FN[config.hidden_act]
+        self.grn = ConvNextV2GRN(4 * dim)
+        self.pwconv2 = nn.Linear(4 * dim, dim)
+        self.drop_path = ConvNextV2DropPath(drop_path) if drop_path > 0.0 else nn.Identity()
+
+    def forward(self, hidden_states: torch.FloatTensor) -> torch.Tensor:
+        input = hidden_states
+        x = self.dwconv(hidden_states)
+        x = x.permute(0, 2, 3, 1)  # (N, C, H, W) -> (N, H, W, C)
```
Comment (sgugger) [inline src/transformers/models/convnextv2/modeling_convnextv2.py:198]: I'd put the comment above and use longer variable names to make it more useful.

File: src/transformers/models/convnextv2/modeling_convnextv2.py [hunk 4]
```diff
@@ -0,0 +1,589 @@
...
+        x = self.layernorm(x)
+        x = self.pwconv1(x)
+        x = self.act(x)
+        x = self.grn(x)
+        x = self.pwconv2(x)
+        x = x.permute(0, 3, 1, 2)  # (N, H, W, C) -> (N, C, H, W)
```
Comment (sgugger) [inline src/transformers/models/convnextv2/modeling_convnextv2.py:204]: Same here.

File: src/transformers/models/convnextv2/modeling_convnextv2.py [hunk 5]
```diff
@@ -0,0 +1,589 @@
...
+        config ([`ConvNextV2Config`]): Model configuration class with all the parameters of the model.
+            Initializing with a config file does not load the weights associated with the model, only the
+            configuration. Check out the [`~PreTrainedModel.from_pretrained`] method to load the model weights.
+"""
+
+CONVNEXTV2_INPUTS_DOCSTRING = r"""
+    Args:
+        pixel_values (`torch.FloatTensor` of shape `(batch_size, num_channels, height, width)`):
+            Pixel values. Pixel values can be obtained using [`AutoImageProcessor`]. See
+            [`AutoImageProcessor.__call__`] for details.
```
Comment (sgugger) [inline src/transformers/models/convnextv2/modeling_convnextv2.py:335]: There is no `AutoImageProcess.__call__`, you need to link to the class of ImageProcessor for this model, so `ConvNextImageProcessor`.

File: tests/test_modeling_common.py [hunk 6]
```diff
@@ -596,6 +596,7 @@ def test_training_gradient_checkpointing(self):
                 or not model_class.supports_gradient_checkpointing
             ):
                 continue
+
```
Comment (sgugger) [inline tests/test_modeling_common.py:599]: There is no need for this change here ;-)

**Review by sgugger — APPROVED** [review]
Thanks for adding this model! Good to go with a few nits on my side.

File: src/transformers/models/convnextv2/modeling_convnextv2.py (hunk 5 above)
Comment (alaradirik) [reply src/transformers/models/convnextv2/modeling_convnextv2.py:335]: Makes sense!

File: src/transformers/models/convnextv2/convert_convnextv2_to_pytorch.py [hunk 7]
```diff
@@ -0,0 +1,282 @@
...
+    model = ConvNextV2ForImageClassification(config)
+    model.load_state_dict(state_dict)
+    model.eval()
+
+    # Check outputs on an image, prepared by ConvNextImageProcessor
+    preprocessor = convert_preprocessor(checkpoint_url)
+    inputs = preprocessor(images=prepare_img(), return_tensors="pt")
+    logits = model(**inputs).logits
+
+    # note: the logits below were obtained without center cropping
```
Comment (amyeroberts) [inline src/transformers/models/convnextv2/convert_convnextv2_to_pytorch.py:172]: Nice comment and checks ❤️

**Review by amyeroberts — APPROVED** [review]
Thanks for adding this model - bonus points for adding the TF model too!

Super nice and clean PR 🤩
